### DISCUSSION TITLE
Adding keymaps for specific surround commands

### DISCUSSION THREAD

**Discussion by firubat** [nvim-mini/mini.nvim#876] [Q&A] [answered]
I'm trying to add keymaps in visual mode of the form `s(` which will add surrounding `()`. I tried the direct approach: 
```
vim.keymap.set({ "x" }, "s(", "sa(", { desc = "Surround selection with ()" })
```
but it results in simply inserting `a(`.

I guess I should call the surround function directly, and not via the keymap `sa`, but I didn't find out how to do it from the docs. Any help appreciated.

**Comment by echasnovski** [comment, marked as answer]
By default `vim.keymap.set()` treats keys in right hand side as if they have default meaning, i.e. no remap, i.e. "noremap".
You can achieve what you want by adding `remap = true` option. Like this:

```
vim.keymap.set({ "x" }, "s(", "sa(", { remap = true, desc = "Surround selection with ()" })
```

------

Just in case: this approach for adding surrounding will work **only** in Visual mode. In Normal mode there needs to be a target (textobject/motion) after `sa`.

**Reply by firubat** [reply]
Thanks!

> Just in case: this approach for adding surrounding will work **only** in Visual mode. In Normal mode there needs to be a target (textobject/motion) after `sa`.

Yes of course, for normal mode I might do some default mapping such as 
```
vim.keymap.set({ "n" }, "s(", "saiW(", { remap = true, desc = "Surround WORD with ()" })
```
